api/graphql/mutations/user.js

```diff
@@ -1,84 +1,194 @@
-import jwt from 'jsonwebtoken'
 import request from 'request'
-import { getPublicKeyFromPem } from '../../../lib/util'
 
-export const createSession = (userId, fetchOne, { req }) => async (root, { email, password }) => {
+// Sign-up Related
+
+export const sendEmailVerification = async (_, { email }) => {
   try {
-    const isLoggedIn = await UserSession.isLoggedIn(req)
-    if (isLoggedIn) {
-      return {
-        me: fetchOne('Me', userId),
-        error: 'already logged-in'
+    let user = await User.find(email, {}, false)
+
+    // Email already validated: 
+    // send failure without error to not reveal account existence
+    if (user?.get('email_validated')) {
+      return { success: false }
```

Comment:
hmm, in this case what do we really want to do? i dont think an error is the right move, if someone is entering an existing email we should send them another verify/login email to finish the process.

Reply:
This is for the case when the email is already validated and so there is no sense in re-validating. What we can do to enhance this particular case is add an `error` key to `GenericResult` and send an error here that is like "trying logging in" or something. I didn't do that yet, and this area can seem a little counter-intuitive in terms of error handling in general, because I was trying to maintain this API endpoint to not be open-game for revealing which emails have accounts.

Reply:
but they may have validated their email and not yet finished registering, in which case we will want to send them another email to get to the registration page, otherwise they would be stuck and not be able to register

Reply:
Ahhh yeah, yeah that makes sense. Will do that. Currently how things will be setup in the front-end if they click on a verify email link and their email is already validated it'll just skip straight to the Registration step so that'll work fine.